autometa/common/external/samtools.py

```diff
@@ -42,49 +41,43 @@ def sort(sam, bam, nproc=mp.cpu_count()):
         </path/to/alignment.sam>
     bam : str
         </path/to/output/alignment.bam>
-    nproc : int, optional
+    cpus : int, optional
         Number of processors to be used. By default uses all the processors of the system
 
     Raises
     ------
     TypeError
-        nproc must be an integer greater than zero
+        cpus must be an integer greater than zero
     FileNotFoundError
         Specified path is incorrect or the file is empty
-    ChildProcessError
-        Samtools did not run successfully, returns the return code from subprocessing call
+    subprocess.CalledProcessError
+        Samtools did not run successfully
     """
 
-    if type(nproc) is not int or nproc <= 0:
-        raise TypeError(f"nproc must be an integer greater than zero! Given: {nproc}")
-    if not os.path.exists(sam) or os.stat(sam).st_size == 0:
+    if type(cpus) is not int or cpus <= 0:
+        raise TypeError(f"cpus must be an integer greater than zero! Given: {cpus}")
+    if not os.path.exists(sam) or not os.path.getsize(sam):
         raise FileNotFoundError(
             f"The specified path: {sam} is either incorrect or the file is empty"
         )
     samtools_out_dir = os.path.dirname(os.path.abspath(bam))
     err = os.path.join(samtools_out_dir, "samtools.err")
     out = os.path.join(samtools_out_dir, "samtools.out")
-    with tempfile.TemporaryDirectory() as tempdir:
-        temp_bam = os.path.join(tempdir, os.path.basename(bam))
+    with tempfile.TemporaryDirectory() as tempdir:  # this will delete the temporary files even if program stops in between
+        sort_fpath = os.path.join(tempdir, os.path.basename(bam))
         cmd = (
-            f"samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {temp_bam}"
+            f"samtools view -@{cpus} -bS {sam} | samtools sort -@{cpus} -o {sort_fpath}"
         )
         logger.debug(f"cmd: {cmd}")
         with open(out, "w") as stdout, open(err, "w") as stderr:
-            retcode = subprocess.call(cmd, stdout=stdout, stderr=stderr, shell=True)
-        if retcode != 0:
-            raise ChildProcessError(f"Sort failed, retcode: {retcode}")
-        else:
-            shutil.move(temp_bam, bam)
+            subprocess.run(cmd, stdout=stdout, stderr=stderr, shell=True, check=True)
+            shutil.move(sort_fpath, bam)
```

Comment:
I think the previous variable name of `temp_bam` is more clear. `samtools sort` outputs a bam file. `sort_fpath` loses this detail. I think the clarity becomes more apparent at the end with `shutil.move(temp_bam, bam)` rather than `shutil.move(sort_fpath, bam)`